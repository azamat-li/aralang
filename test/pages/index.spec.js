import Vuetify from 'vuetify'

// Components
import index from '@/pages/index'

// Utilities
import {createLocalVue, RouterLinkStub, shallowMount} from '@vue/test-utils'

describe('index.vue', () => {
  const localVue = createLocalVue()
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  it('should have a custom title and match snapshot', () => {
    const wrapper = shallowMount(index, {
      localVue,
      vuetify,
      stubs: {
        NuxtLink: RouterLinkStub
      }
    })

    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot()

    // We could also verify this differently
    // by checking the text content
    const title = wrapper.find(' .intro')

    expect(title.text()).toBe('Изучай арабский на своём уровне')
  })

})
