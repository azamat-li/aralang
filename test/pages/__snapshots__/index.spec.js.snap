// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.vue should have a custom title and match snapshot 1`] = `
"<v-row-stub tag=\\"div\\" align=\\"center\\" justify=\\"center\\">
  <v-col-stub cols=\\"12\\" sm=\\"8\\" md=\\"6\\" tag=\\"div\\">
    <div class=\\"text-center\\"></div>
    <v-card-stub loaderheight=\\"4\\" tag=\\"div\\">
      <v-card-title-stub class=\\"headline\\"><strong>Ara</strong>bic <strong>lang</strong>uage
      </v-card-title-stub>
      <v-card-text-stub>
        <p class=\\"intro\\">Изучай арабский на своём уровне</p>
        <p>
          For more information on Vuetify, check out the
          <a href=\\"https://vuetifyjs.com\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">
            documentation </a>.
        </p>
        <p>
          If you have questions, please join the official
          <a href=\\"https://chat.vuetifyjs.com/\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" title=\\"chat\\">
            discord </a>.
        </p>
        <p>
          Find a bug? Report it on the github
          <a href=\\"https://github.com/vuetifyjs/vuetify/issues\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" title=\\"contribute\\">
            issue board </a>.
        </p>
        <p>
          Thank you for developing with Vuetify and I look forward to bringing
          more exciting features in the future.
        </p>
        <div class=\\"text-xs-right\\"><em><small>— John Leider</small></em></div>
        <hr class=\\"my-3\\"> <a href=\\"https://nuxtjs.org/\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">
          Nuxt Documentation
        </a> <br> <a href=\\"https://github.com/nuxt/nuxt.js\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">
          Nuxt GitHub
        </a>
      </v-card-text-stub>
      <v-card-actions-stub>
        <v-spacer-stub></v-spacer-stub>
        <v-btn-stub color=\\"primary\\" tag=\\"button\\" activeclass=\\"\\" to=\\"/inspire\\" nuxt=\\"true\\" type=\\"button\\"> Continue </v-btn-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-col-stub>
</v-row-stub>"
`;
